lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -632,17 +632,25 @@ class BroadcastOpConversionPattern
     // ops that support implicit broadcasting. We expect each Op's verify
     // function to assert their arguments to verify that they can broadcast.
 
-    mlir::Value input = srcOp.getOperand(0);
-    mlir::Value result = srcOp.getResult();
-
     if (srcOp->getUsers().empty()) {
-      return rewriter.notifyMatchFailure(
-          srcOp, "ttir.broadcast op should have at least one use.");
+      rewriter.eraseOp(srcOp);
+      return success();
     }
 
-    rewriter.replaceAllUsesWith(result, input);
-    rewriter.eraseOp(srcOp);
+    // This solution only works for one level of cascading broadcast Ops, it
+    // can be extended to allow for any levels of cascading Broadops.
+    // uazizTT: TODO: Extend the solution to work on multi-level cascading
+    // Broadcast Ops.
 
+    mlir::Value input = srcOp.getOperand(0);
+    auto nextOp = *srcOp->getUsers().begin();
+    if (isa<ttir::BroadcastOp>(nextOp)) {
+      rewriter.replaceAllOpUsesWith(nextOp, input);
+    } else {
+      rewriter.replaceAllOpUsesWith(srcOp, input);
+    }
```

Comment:
Haha nice workaround :) You'd just have to iterate over all srcOp users and do this, before erasing the srcOp.

Since this will only work for 2 levels, and seems like it would be cumbersome to cover more levels like this, I think at this point it might be better maybe to have a pass which does a walk through each module region/block, and just calls `replaceAllOpUsesWith` on each broadcast op inside the region/block. That shouldn't cause issues with editing the other ops like it does in separate matchAndRewrite for ops.

Reply:
Sure, but I think it will be equally cumbersome, I just updated with the solution for multiple level cascading.